php/commands/db.php

```diff
@@ -363,27 +366,23 @@ public function import( $args, $assoc_args ) {
 			$result_file = sprintf( '%s.sql', DB_NAME );
 		}
 
-		if ( '-' === $result_file ) {
-			$descriptors = array(
-				STDIN,
-				STDOUT,
-				STDERR,
-			);
-		} else {
-			if ( ! file_exists( $result_file ) ) {
-				WP_CLI::error( sprintf( 'Import file missing: %s', $result_file ) );
+		$mysql_args = array(
+			'database' => DB_NAME,
+		);
+
+		if ( '-' !== $result_file ) {
+			if ( ! is_readable( $result_file ) ) {
+				WP_CLI::error( sprintf( 'Import file missing or not readable: %s', $result_file ) );
 			}
 
-			$descriptors = array(
-				array( 'file', $result_file, 'r' ),
-				STDOUT,
-				STDERR,
-			);
+			$query = \WP_CLI\Utils\get_flag_value( $assoc_args, 'skip-optimization' )
+				? 'SOURCE %s;'
```

Comment:
@schlessera if the filename is `-` then the mysql call does not include any query to execute, but prompts the CLI.